AppCenter/AppCenter/Internals/Channel/MSChannelGroupDefault.m

```diff
@@ -177,6 +186,17 @@ - (void)channel:(id<MSChannelProtocol>)channel didResumeWithIdentifyingObject:(i
 
 - (void)setEnabled:(BOOL)isEnabled andDeleteDataOnDisabled:(BOOL)deleteData {
 
+#if !TARGET_OS_OSX
+  if (isEnabled) {
+    [MS_NOTIFICATION_CENTER addObserver:self
```

Comment:
We don't really know when the `applicationWillTerminate` of the developer will be called. He can use the delegate but also the notification as we do. Could be before or after. If it's happening after would the log be lost?

Reply:
yes

Reply:
Could be. One possible solution is to use a special flag inside our callback and handle it later but seems like a bit of overkill. What do you think?

Reply:
Sounds overkill yes. I wonder if we can simply switch to synchronous storage for all logs when the app enters this phase. The app will be wiped out of the memory so we don't need to think about restoring the async storage. We'll need to make sure previous logs are saved.

If it turns out to introduce too much complexity we'll need to postpone the fix.